modules/account-lib/src/coin/dot/transactionBuilder.ts

```
@@ -58,15 +58,15 @@ export abstract class TransactionBuilder extends BaseTransactionBuilder {
    *
    * The number of blocks after the checkpoint for which a transaction is valid. If zero, the transaction is immortal.
    *
-   * @param {number | undefined} eraPeriod
+   * @param {number | undefined} maxDuration
    * @returns {TransactionBuilder} This transaction builder.
    *
    * @see https://wiki.polkadot.network/docs/build-transaction-construction
    */
-  eraPeriod(eraPeriod: number | undefined): this {
-    if (eraPeriod !== undefined) {
-      this.validateValue(new BigNumber(eraPeriod));
-      this._eraPeriod = eraPeriod;
+  durationConfig({ maxDuration }: { maxDuration: number | undefined }): this {
+    if (maxDuration !== undefined) {
```

maybe throw an error in else condition?

---

we dont need to as this value is optional